Makefile

```
@@ -42,48 +50,58 @@ patch-dev: superset-clean
 	ln -s "$(PWD)/$(PATCH_SOURCE_DIR)/superset/superset_config_dev.py" "$(SUPERSET_DIR)/superset_config.py"; \
 
 # Create docker image
-.PHONY: superset-build
-superset-build: patch
-	docker build -t $(SUPERSET_IMAGE_NAME):$(VERSION) -f superset/contrib/docker/Dockerfile superset
+.PHONY:
+build: patch
+	docker build -t $(DOCKER_IMAGE_NAME):$(VERSION) -f superset/contrib/docker/Dockerfile $(SUPERSET_DIR)
 
-# Push the superset docker image, based on .ci/Makefile.main
-superset-docker-push: docker-login superset-build
+docker-validate:
+	@if [ -z "$(DOCKER_USERNAME)" ]; then \
+		echo "DOCKER_USERNAME variable cannot be empty."; \
+		exit 1; \
+	fi; \
+	if [ -z "$(DOCKER_PASSWORD)" ]; then \
+		echo "DOCKER_PASSWORD variable cannot be empty."; \
+		exit 1; \
+	fi
+
+docker-login: docker-validate
+	@docker login -u "$(DOCKER_USERNAME)" -p "$(DOCKER_PASSWORD)" $(DOCKER_REGISTRY); \
+
+# Push the docker image
+docker-push: docker-login build
 	@if [ "$(BRANCH)" == "master" && "$(DOCKER_PUSH_MASTER)" == "" ]; then \
 		echo "docker-push is disabled on master branch" \
 		exit 1; \
 	fi; \
-	docker push $(SUPERSET_IMAGE_NAME):$(VERSION); \
-	if [ -n "$(DOCKER_PUSH_LATEST)" ]; then \
-		docker tag $(SUPERSET_IMAGE_NAME):$(VERSION) \
-			$(SUPERSET_IMAGE_NAME):latest; \
-		docker push $(SUPERSET_IMAGE_NAME):latest; \
+	docker push $(DOCKER_IMAGE_NAME):$(VERSION); \
+	if [ -n "$(DOCKER_IMAGE_NAME)" ]; then \
+		docker tag $(DOCKER_IMAGE_NAME):$(VERSION) \
+			$(DOCKER_IMAGE_NAME):latest; \
+		docker push $(DOCKER_IMAGE_NAME):latest; \
 	fi;
 
-superset-docker-push-latest-release:
-	@DOCKER_PUSH_LATEST=$(IS_RELEASE) make superset-docker-push
+docker-push-latest-release:
+	@DOCKER_PUSH_LATEST=$(IS_RELEASE) make docker-push
 
 # Clean superset directory from copied files
-.PHONY: superset-clean
-superset-clean:
+.PHONY:
```

should be this instead?
```suggestion
.PHONY: clean
```

---

thanks. fixed. Also fixed some other targets that didn't have `.PHONY`